### DISCUSSION TITLE
Self hosted Giscus throws "Cannot use giscus on this repository. Make sure all of the above criteria has been met." though repo has permission

### DISCUSSION THREAD

**Discussion by tcnaganath** [giscus/giscus#1550] [Q&A] [answered]
Self hosted Giscus throws "Cannot use giscus on this repository. Make sure all of the above criteria has been met." though repo has permissions to the self hosted app. In nginx logs it is found that 403 error. Please let me know how to resolve this issue. All the steps in https://github.com/giscus/giscus/blob/main/SELF-HOSTING.md has been followed except supabase which is optional.

**Comment by laymonage** [comment, marked as answer]
Check your private key configuration, and also check the Next.js server logs.

**Reply by tcnaganath** [reply]
Thanks @laymonage. That works. One more question, I wanted to redirect all the compiled files to the <domain_server>/giscus/ /site.webmanifest by default giscus app directs to <domain_server>/site.webmanifest. Where we need to make the changes to redirect all the compiled files like js, css, img files to <domain_server>/giscus/

**Reply by laymonage** [reply]
If you search the code you should be able to find it https://github.com/giscus/giscus/blob/5daac0147242a05408811328385eec6e61dacb14/pages/_document.tsx#L39-L40

**Reply by tcnaganath** [reply]
Thanks @laymonage

**Reply by tcnaganath** [reply]
@laymonage In order to set basePath to redirect the static js files to the basePath "<domain_name>/giscus/_next/static/chunks/**.js instead of <domain_server>/_next/static/chunks/**.js. what needs to be changed? Any suggestions

**Reply by laymonage** [reply]
https://nextjs.org/docs/app/api-reference/next-config-js/assetPrefix

**Reply by tcnaganath** [reply]
@laymonage I have tried adding in next.config.js but still didnt work.

```
const config = withBundleAnalyzer(
  withPreact(
    nextTranslate({
      basePath: '/giscus',
      assetPrefix: '/giscus/',
      async headers() {
        return [
          {
            source: '/(.*)',
            headers: securityHeaders,
          },
          {
            source: '/',
            headers: [
              {
                key: 'X-Frame-Options',
                value: 'SAMEORIGIN',
              },
            ],
          },
          {
            source: '/(themes/(?:.*)|client\\.js)',
            headers: [
              {
                key: 'Cache-Control',
                value: `public, max-age=0, stale-while-revalidate=${swr}`,
              },
            ],
          },
        ];
      },
      experimental: {
        browsersListForSwc: true,
        legacyBrowsers: false,
        esmExternals: false,
      },
    }),
  ),
);
```

**Comment by shayanhoshyari** [comment]
same happened to me.

in my case issue was the variable ORIGIN was not set on the server

when that is not set, the code related to CORS throws and the server returns 500 if the Origin header in the request is not set. When the onboarding UI calls the server, it does bot set the Origin header.

after I fixed above, the. I had to fix the private key, I was missing the line breaks after —-

what was very helpful was running the server locally first and replicating the browser calls with curl
